packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -152,7 +155,7 @@ export interface IGarbageCollector {
         options: { logger?: ITelemetryLogger; runGC?: boolean; runSweep?: boolean; fullGC?: boolean; },
     ): Promise<IGCStats>;
     /** Summarizes the GC data and returns it as a summary tree. */
-    summarize(): ISummaryTreeWithStats | undefined;
+    summarize(fullTree: boolean, trackState: boolean): ISummaryTreeWithStats | undefined;
```

Comment:
Not sure if this is what we want, but this logic lives in the `SummarizerNode` plus the `SummarizerNodeWithGC`.

Reply:
I'm not sure what's the best approach here (I'd say I know less about layers involved than you and Navin).
I'll only point out that SummarizerNode flow (i.e. decision made by SummarizerNode if a node should re-summarize or re-use) is based on a signal that something changed under node. And we do not have that signal.

We should strongly consider more general flow of blob dedupping across runtime. We had this code in ODSP driver, but it was super buggy, and not factored properly as separate layer, so we removed it.
Sooner or later, we will arrive to a conclusion that we need this capability. I.e. many DDSs will have similar problem you are facing here, and they could solve it by carefully tracking state of DDS across summary blobs and making decisions if blob should be regenerated or not - that's the most efficient way to do it, but the most complicated and error prone. Alternative - always generate blobs and let central generic system deduplicate blobs against blobs in prior summary.
I'd write such mechanism as an adapter layers, similar to BlobAggregationStorage

Reply:
I'd rather sort it :) I think it will be much easier to deal with it if blob content is reliable.
That said, what influences randomness? I.e., what events / processes should result in shuffling of that data? Does reoder happen often enough where we start caring? Are these events / effects avoidable (or their effects can be neutralized)?

Reply:
I think we should sort it. That will make the comparison pretty much a string compare. You will have to sort both the nodes and the outboundRoutes array. It may be a little expensive to do so but I think its worth it.
As for what causes this randomness, it can be any number of factors like what Tyler pointed out. The outbound routes for instance that are added by DDSs are controlled by them and we can't rely on what they do. So, GarbageCollector should sort them before writing them in the summary.